### ISSUE TITLE
Embeds can't have 2 fields with the same name

### ISSUE THREAD

**Issue by 1PV1** [discord-php/DiscordPHP#1142]
**Environment**
- PHP Version:
    - 8.2.7
- DiscordPHP Version:
    - dev-master c892cfb

**Describe the bug**
Embeds can't have 2 fields with the same name because `fields` in an embed is a collection, we should be able to create multiple fields with the same name (`\u{200b}` for example).
A good solution would be to make `fields` an array.

**To Reproduce**
```php
// Has one field
print_r((new Embed($discord))->setColor(0xffffff)->addField(...[[ "name" => "\u{200b}", 'value' => "\u{200b}", "inline" => false ], [ "name" => "\u{200b}", 'value' => "\u{200b}", "inline" => false ]]));

// Has two fields
print_r((new Embed($discord))->setColor(0xffffff)->addField(...[[ "name" => "\u{200b}", 'value' => "\u{200b}", "inline" => false ], [ "name" => "test", 'value' => "\u{200b}", "inline" => false ]]));
```

**Comment by key2peace** [comment]
This limit is on the discord side, we do not check for duplicate names.
[Code reference](https://github.com/discord-php/DiscordPHP/blob/master/src/Discord/Parts/Embed/Embed.php#L304-L348)

**Comment by key2peace** [comment]
actually looking at [this](https://github.com/discord-php/DiscordPHP/blob/master/src/Discord/Parts/Embed/Embed.php#L140-L157) now..

**Comment by SQKo** [comment]
Thanks for reporting, should be fixed in [`dev-master#9fab769`](https://github.com/discord-php/DiscordPHP/commit/9fab769aa47a57052f6fa0309bcede6abe81e0ca)
